tests/test_hinfo.py

```diff
@@ -1,51 +1,40 @@
 import os
-import shutil
-import sqlite3
-import tempfile
 from pathlib import Path
 
+import lsst.utils
 import pytest
+import sqlalchemy as sa
 import yaml
-from lsst.consdb import hinfo, utils
+from felis.datamodel import Schema
+from felis.db.utils import DatabaseContext
+from felis.metadata import MetaDataBuilder
+from felis.tests.postgresql import setup_postgres_test_db
+from lsst.consdb import hinfo
 from lsst.resources import ResourcePath
-from sqlalchemy import MetaData, Table, select
+from sqlalchemy import MetaData, select
 
 
 @pytest.fixture
-def tmpdir(scope="module"):
-    os.environ["INSTRUMENT"] = "LATISS"
-    tmpdir = Path(tempfile.mkdtemp())
-    try:
-        yield tmpdir
-    finally:
-        shutil.rmtree(tmpdir)
+def pg_engine(request, scope="module"):
+    schema_name = request.param if hasattr(request, "param") else "cdb_latiss"
 
+    schema_file = os.path.join(lsst.utils.getPackageDir("sdm_schemas"), "yml", schema_name + ".yaml")
+
+    with open(schema_file) as f:
+        yaml_data = yaml.safe_load(f)
+    schema = Schema.model_validate(yaml_data)
+    md = MetaDataBuilder(schema).build()
+
+    with setup_postgres_test_db() as instance:
+        context = DatabaseContext(md, instance.engine)
+        print(f"{type(instance.engine)=}")
+        context.initialize()
+        context.create_all()
+
+        hinfo.engine = instance.engine
+        hinfo.instrument = "LATISS"
 
-@pytest.fixture
-def engine(tmpdir, scope="module"):
-    db_path = tmpdir / "test.db"
-    instruments = ["latiss"]
-    with sqlite3.connect(db_path) as conn:
-        conn.execute("CREATE TABLE schemas (name text, path text)")
-        for instrument in instruments:
-            schema = f"cdb_{instrument}"
-            sql = Path(__file__).parent / f"{schema}.sql"
-            if not sql.exists():
-                continue
-            schema_path = tmpdir / f"{schema}.db"
-            conn.execute(f"INSERT INTO schemas VALUES ('{schema}', '{schema_path}')")
-            with sqlite3.connect(schema_path) as schema_conn:
-                schema_conn.executescript(sql.read_text())
-            schema_conn.close()
-
-    os.environ["POSTGRES_URL"] = f"sqlite:///{db_path}"
-    hinfo.engine = utils.setup_postgres()
-    hinfo.instrument = "LATISS"
-
-    try:
         yield hinfo.engine
-    finally:
```

Comment:
Why remove the try/finally?

Reply:
No longer needed because the "with setup_postgres_test_db()" handles teardown.